Tests/Microsoft.AppCenter.Crashes.Test.Windows/HandledErrorTest.cs

```diff
@@ -0,0 +1,237 @@
+                It.IsAny<IDictionary<string, string>>(),
+                It.IsAny<string>(),
+                It.IsAny<CancellationToken>()))
+                .Callback((string uri, string method, IDictionary<string, string> headers, string content, CancellationToken cancellation) =>
+                {
+                    actualLog = JsonConvert.DeserializeObject<LogContainer>(content, LogSerializer.SerializationSettings).Logs.Single() as HandledErrorLog;
+                    semaphore.Release();
+                });
+            var exception = new System.Exception("Something went wrong.");
+            var maxProperties = 20;
+            var properties = new Dictionary<string, string>();
+            for (int i = 0; i < maxProperties + 5; ++i)
+            {
+                properties[i.ToString()] = i.ToString();
+            }
+            Crashes.TrackError(exception, properties);
+
+            // Wait until the http layer sends the log.
+            semaphore.Wait(1000);
+            Assert.IsNotNull(actualLog);
+            Assert.AreEqual(exception.Message, actualLog.Exception.Message);
+            CollectionAssert.IsSubsetOf(actualLog.Properties as Dictionary<string, string>, properties);
+            Assert.AreEqual(maxProperties, actualLog.Properties.Count);
+        }
+
+        [TestMethod]
+        public void TrackErrorWithoutProperties()
+        {
+            var mockHttpNetworkAdapter = StartCrashes();
+            var semaphore = new SemaphoreSlim(0);
+            HandledErrorLog actualLog = null;
+            Mock.Get(mockHttpNetworkAdapter).Setup(adapter => adapter.SendAsync(
+                    It.IsAny<string>(),
+                    It.IsAny<string>(),
+                    It.IsAny<IDictionary<string, string>>(),
+                    It.IsAny<string>(),
+                    It.IsAny<CancellationToken>()))
+                .Callback((string uri, string method, IDictionary<string, string> headers, string content, CancellationToken cancellation) =>
+                {
+                    actualLog = JsonConvert.DeserializeObject<LogContainer>(content, LogSerializer.SerializationSettings).Logs.Single() as HandledErrorLog;
+                    semaphore.Release();
+                });
+            var exception = new System.Exception("Something went wrong.");
+            Crashes.TrackError(exception);
+
+            // Wait until the http layer sends the log.
+            semaphore.Wait(1000);
+            Assert.IsNotNull(actualLog);
+            Assert.AreEqual(exception.Message, actualLog.Exception.Message);
+            Assert.IsNull(actualLog.Properties);
+        }
+
+        [TestMethod]
+        public void TrackErrorWithoutException()
+        {
+            var mockHttpNetworkAdapter = StartCrashes();
+            var semaphore = new SemaphoreSlim(0);
+            HandledErrorLog actualLog = null;
+            Mock.Get(mockHttpNetworkAdapter).Setup(adapter => adapter.SendAsync(
```

Comment:
The test scenario doesn't require this mock unless you fail the test in the callback.

Reply:
Can you elaborate? Seems like it is necessary to capture the log that is enqueued.

Reply:
Is `SendAsync` really called for `null` exception?

Reply:
No, but that is what this test is verifying

Reply:
Assert fail in the callback would be a good option to resolve this comment.

Reply:
Done.